code/client/Dockerfile

```diff
@@ -1,6 +1,27 @@
-FROM nginx:1.15.10-alpine
-COPY default.conf /etc/nginx/conf.d/
-COPY dist/ /usr/share/nginx/html/
+FROM node:latest AS builder
 
+# Set working directory
+WORKDIR /app
 
+# Copy all files from current directory to working dir in image
+COPY . .
 
+# install node modules and build assets
+RUN npm install -g @angular/cli
+
+RUN bash build.sh
+
+# nginx state for serving content
+FROM nginx:alpine
+
+# Set working directory to nginx asset directory
+WORKDIR /usr/share/nginx/html
+
+# Remove default nginx static assets
+RUN rm -rf ./*
+
+# Copy static assets from builder stage
+COPY --from=builder /app/dist/client .
+
+# Containers run nginx with global directives and daemon off
+ENTRYPOINT ["nginx", "-g", "daemon off;"]
```

Comment:
Hi @HarshCasper. Sorry, Asmi just brought my attention to this. I'm trying to think this through. Technically, the reporting tool functionality is primarily carried out in the backend - so i put more weight on the backend development / testing. I didn't  do as much on the way of testing the client as we primarily needed it as a demo at that time. This can however be improved on by myself / the community. What we have to figure is where the api will be hosted for the purpose of testing the integrations.

Reply:
Okay. I have added the requested changes @HarshCasper you can review it then.